### ISSUE TITLE
Start Minikube with rkt on OSX

### ISSUE THREAD

**Issue by johscheuer** [kubernetes/minikube#603]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

I tried to start Minikube with rkt as container engine as described in the Readme:

```
$ minikube start --vm-driver="xhyve" --memory=4096 --cpus=2 \  
    --network-plugin=cni \
    --container-runtime=rkt \
    --iso-url=https://github.com/coreos/minikube-iso/releases/download/v0.0.3/minikube-v0.0.3.iso
Starting local Kubernetes cluster...
E0916 18:09:20.608688   30091 start.go:85] Error starting host: Error creating host: Error creating machine: Error in driver during machine creation: open /Users/jscheuermann/.minikube/machines/minikube/b2d-image/boot/vmlinuz64: no such file or directory. Retrying.
E0916 18:11:21.994860   30091 start.go:85] Error starting host: Error starting stopped host: Machine didn't return an IP after 120 seconds, aborting. Retrying.


$ ll /Users/jscheuermann/.minikube/machines/minikube/b2d-image/boot/
total 133176
-rw-r--r--  1 jscheuermann  staff   6.2M Sep  8 14:05 bzImage
-rw-r--r--  1 jscheuermann  staff    59M Sep  8 14:05 initrd
```

This also results in a read-only mount:

```
 mount | grep minikube
/dev/disk3 on /Users/jscheuermann/.minikube/machines/minikube/b2d-image (cd9660, local, nodev, nosuid, read-only, noowners)
```

After this I need to clean up manually:

```
sudo umount  /Users/jscheuermann/.minikube/machines/minikube/b2d-image && rm -rf Users/jscheuermann/.minikube
```

**Minikube version** (use `minikube version`):
10.0.0

**Environment**:
- **OS** (e.g. from /etc/os-release):
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): "DriverName": "xhyve",
- **Docker version** (e.g. `docker -v`): Docker version 1.12.0, build 8eab29e
- **Install tools**:
- **Others**:

**What happened**:
Cluster didn't came up and the mount wasn't cleaned up

**What you expected to happen**:
Start a Kubernetes cluster with rkt as container engine

**How to reproduce it** (as minimally and precisely as possible):

```
minikube start --vm-driver="xhyve" --memory=4096 --cpus=2 \  
    --network-plugin=cni \
    --container-runtime=rkt \
    --iso-url=https://github.com/coreos/minikube-iso/releases/download/v0.0.3/minikube-v0.0.3.iso
```

**Anything else do we need to know**:

I tried also running it with Virtualbox (without success): `5.0.26`
OSX: `15.6.0 Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64 x86_64`

This is related with: https://github.com/kubernetes/minikube/issues/599 but if the cluster creation fails the mount should be clean up?

**Comment by aaron-prindle** [comment]
Thanks for the detailed report, sorry you had trouble with the xhyve driver and rkt.   There is a known issue using xhyve with the rkt iso that needs to be addressed upstream (see the referenced issue above).  You can get the rkt iso working with vbox though: if you try a  `minikube delete` and `minikube start` (using the default Virtualbox Driver) it should work.

**Comment by r2d4** [comment]
This has been fixed in the latest xhyve driver 0.3.1
